lib/react.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const React = require('react');
+
+const Context = React.createContext();
+
+Context.displayName = 'StrangeMiddleEnd';
+
+exports.Provider = ({ middleEnd, children }) => {
+
+    if (!middleEnd || !middleEnd.initialized) {
```

Comment:
What do you think about making the assertion this way versus with prop types?  Again, I am sort of looking to react-redux for inspiration.

Reply:
To be clear— I am totally down with this if we like the behavior, just looking for thoughts/considerations.

Reply:
Oh man, yea, I flip-flopped on this, initially thought about doing something like react-redux's proptypes approach. 

I like throwing in the abstract, but yea, I'm guessing that's less idomatic React than using proptypes and I'm shoehorning expectations from non-React work onto this situation, which might catch more experienced, informed React users off guard.  Though if the end result is a component trying to use the uninit'd middle-end throwing anyway, possibly somewhere that's not immediately apparent e.g. not in the app's initial view, so the app could load fine, maybe giving a false sense of security, throwing earlier seems more likely to catch the programmer's eye than a proptypes warning. maybe?

So if I had to vote, I'd vote for throwing. But there's a minimally (at best) informed rationale behind that vote  🤡

Reply:
Cool, thanks for that run down!  Let's go for the throw.  We can always change it!